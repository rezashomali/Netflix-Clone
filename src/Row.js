import React, { useState, useEffect } from "react";
import axios from "./axios";
import YouTube from "react-youtube";
import movieTrailer from "movie-trailer";
import "./Row.css";

const base_url = "https://image.tmdb.org/t/p/original/";

function Row({ title, fetchUrl, isLargeRow }) {
  const opts = {
    height: "390",
    width: "100%",
    playerVars: {
      autoplay: 1,
    },
  };

  const [movies, setMovies] = useState([]);
  const [trailerUrl, setTrailerUlr] = useState("");

  useEffect(() => {
    async function fetchData() {
      const request = await axios.get(fetchUrl);
      setMovies(request.data.results);
      // console.log(request.data.results);
    }
    fetchData();
  }, [fetchUrl]);

  const handleClick = (movie) => {
    if (trailerUrl) {
      setTrailerUlr("");
    } else {
      movieTrailer(movie?.name || movie?.original_name || movie?.original_title)
        .then((url) => {
          const urlParams = new URLSearchParams(new URL(url).search);
          console.log(urlParams.get("v"));
          setTrailerUlr(urlParams.get("v"));
        })
        .catch((error) => console.log(error));
    }
  };

  return (
    <div className="row">
      <h2>{title}</h2>

      <div className="row__posters">
        {movies.map((item) => (
          <img
            onClick={() => handleClick(item)}
            key={item.id}
            className={`row__poster ${isLargeRow && "row_posterLarge"}`}
            src={
              base_url + (isLargeRow ? item.poster_path : item.backdrop_path)
            }
            alt={item.original_title}
          />
        ))}
      </div>

      {trailerUrl && <YouTube videoUrl={trailerUrl} opts={opts} />}
    </div>
  );
}

export default Row;
