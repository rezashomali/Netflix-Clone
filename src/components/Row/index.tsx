import React, { useState, useEffect } from "react";
import axios from "../../Api/axios";
import YouTube from "react-youtube";
import "./Row.css";
// import movieTrailer from "movie-trailer";
const movieTrailer = require("movie-trailer");

const base_url: string = "https://image.tmdb.org/t/p/original/";

interface Props {
  title: string;
  fetchUrl: string;
  isLargeRow?: boolean;
}

interface MovieType {
  backdrop_path: string;
  name?: string;
  title?: string;
  original_name?: string;
  overview: string;
  map: any;
}

const YouTubeOption = {
  height: "390",
  width: "100%",
};

const Row: React.FC<Props> = ({ title, fetchUrl, isLargeRow }) => {
  const [movies, setMovies] = useState<MovieType>();
  const [trailerUrl, setTrailerUlr] = useState<string | null>("");

  useEffect(() => {
    async function fetchData() {
      const request = await axios.get(fetchUrl);
      setMovies(request.data.results);
      // console.log(request.data.results);
    }
    fetchData();
  }, [fetchUrl]);

  const handleClick = (movie: {
    name?: string;
    original_name?: string;
    original_title?: string;
  }) => {
    if (trailerUrl) {
      setTrailerUlr("");
    } else {
      movieTrailer(movie?.name || movie?.original_name || movie?.original_title)
        .then((url: string) => {
          const urlParams = new URLSearchParams(new URL(url).search);
          setTrailerUlr(urlParams.get("v"));
        })
        .catch((error: string) => console.log(error));
    }
  };

  return (
    <div className="row">
      <h2>{title}</h2>

      <div className="row__posters">
        {movies?.map(
          (item: {
            id: number;
            original_title: string;
            poster_path: string;
            backdrop_path: string;
          }) => (
            <img
              onClick={() => handleClick(item)}
              key={item.id}
              className={`row__poster ${isLargeRow && "row_posterLarge"}`}
              src={
                base_url + (isLargeRow ? item.poster_path : item.backdrop_path)
              }
              alt={item.original_title}
            />
          )
        )}
      </div>

      {trailerUrl && <YouTube videoId={trailerUrl} opts={YouTubeOption} />}
    </div>
  );
};

export default Row;
