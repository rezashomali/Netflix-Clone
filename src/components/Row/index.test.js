import React from "react";
import renderer from "react-test-renderer";
import Row from "./index";

test("Row snapshot ", () => {
  const component = renderer.create(<Row title="" fetchUrl={""} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
