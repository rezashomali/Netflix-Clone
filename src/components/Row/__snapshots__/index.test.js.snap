// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row snapshot  1`] = `
<div
  className="row"
>
  <h2>
    
  </h2>
  <div
    className="row__posters"
  />
  
</div>
`;
